providers/dns/exoscale/exoscale.go

```diff
@@ -84,12 +81,8 @@ func NewDNSProviderConfig(config *Config) (*DNSProvider, error) {
 		return nil, errors.New("exoscale: credentials missing")
 	}
 
-	egoscale.UserAgent = "go-acme/lego " + egoscale.UserAgent
```

Comment:
Any reason for dropping it?

Reply:
Because it's a global variable, I don't want to produce side effects: if someone uses lego and also exocale as a lib, using the global variable will be a problem.

Generally speaking, the use of global variables should be avoided.

Reply:
IMHO, the `User-Agent` should be an option of the `Client` like the other `ClientOpt` instead of a global variable.

Reply:
I was hoping something like that: https://github.com/exoscale/egoscale/compare/master...ldez:egoscale:feat/user-agent

I will not open a PR because it's a breaking change, except if you want it.